src/game_over/mod.rs

```diff
@@ -0,0 +1,68 @@
+use bevy::prelude::*;
+
+use crate::{
+    pause_menu::button::{UIButton, HOVERED_BUTTON, NORMAL_BUTTON, PRESSED_BUTTON},
+    states::GameStates,
+};
+
+use self::ui::GameOverScreenComponent;
+
+pub struct GameOverPlugin;
+
+mod ui;
+
+impl Plugin for GameOverPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system_set(SystemSet::on_enter(GameStates::GameOver).with_system(ui::build_ui));
+        app.add_system_set(SystemSet::on_update(GameStates::GameOver).with_system(button_handler));
+        app.add_system_set(SystemSet::on_exit(GameStates::GameOver).with_system(ui_destroyer));
+
+        app.add_system_set(
+            SystemSet::on_update(GameStates::Main).with_system(open_gameover_screen),
+        );
+    }
+}
+
+fn open_gameover_screen(
+    mut keyboard: ResMut<Input<KeyCode>>,
+    mut game_state: ResMut<State<GameStates>>,
+) {
+    if keyboard.just_pressed(KeyCode::G) {
+        game_state.push(GameStates::GameOver).unwrap();
+        keyboard.reset(KeyCode::G);
+    }
+}
+
+fn ui_destroyer(mut commands: Commands, query: Query<Entity, With<GameOverScreenComponent>>) {
```

Comment:
Should we also make sure we attach a `GameOverScreenComponent` to the player to avoid the duplication issue we're seeing when restarting the game?

Reply:
Or just query for entities with a player component and despawn those.

Reply:
not at all, it should be a completely separate destructor and component